packages/ckeditor5-utils/src/version.ts

```diff
@@ -24,138 +24,162 @@ declare global {
+	 * in the exact same version.
+	 *
+	 * If you use third-party plugins, make sure to update them, too. If they are not compatible
+	 * with the version of CKEditor&nbsp;5 you use, you may need to downgrade the CKEditor&nbsp;5 packages
+	 * (which we do not recommend). Ask the author of said plugin to upgrade the dependencies,
+	 * or fork their project and update it yourself.
+	 *
+	 * # Check imports
+	 *
+	 * The next step is to look at how you import CKEditor&nbsp;5 into your project.
+	 *
+	 * **The {@glink updating/nim-migration/migration-to-new-installation-methods new installation methods}
+	 * are designed to prevent module duplication, so if you are not using them yet, you should consider
+	 * updating your project**. However, for backwards compatibility, a number of legacy installation methods
+	 * is still supported, and mixing them may result in module duplication.
+	 *
+	 * These are the most common import methods of the CKEditor&nbsp;5 packages.
+	 *
+	 * - **New installation methods (NIM)** &ndash; Imports from the `ckeditor5` and `ckeditor5-premium-features` packages.
+	 * - **Optimized build** for the new installation methods &ndash; Imports from the `@ckeditor/ckeditor5-<NAME>/dist/index.js`.
+	 * - **Predefined builds** (legacy) &ndash; Imports from the `@ckeditor/ckeditor5-build-<NAME>` packages.
+	 * - **Default imports** (legacy) &ndash; Imports from the `@ckeditor/ckeditor5-<NAME>` packages (default export).
+	 * - **`src`** (legacy) &ndash; Imports from the `@ckeditor/ckeditor5-<NAME>/src/*`.
+	 * - **DLL builds** (legacy) &ndash; Imports from the `ckeditor5/build/<NAME>` and `@ckeditor/ckeditor5-<NAME>/build/*`.
+	 *
+	 * The best way to avoid duplicate modules is to not mix these installation methods. For example, if you use imports
+	 * specific to the optimized build, you should use them for all CKEditor&nbsp;5 packages. In addition, since
+	 * the Predefined and DLL builds already include the core of the editor, they cannot be used with other types of imports.
+	 *
+	 * This is a matrix showing which installation methods are compatible with each other:
+	 *
+	 * |                  | NIM | Optimized build | Predefined builds | Default imports | `src` | DLL builds |
+	 * |------------------|-----|-----------------|-------------------|-----------------|-------|------------|
+	 * | NIM              | ✅  | ✅              | ❌                | ❌              | ❌    | ❌         |
+	 * | Optimized builds | ✅  | ✅              | ❌                | ❌              | ❌    | ❌         |
+	 * | Predefined build | ❌  | ❌              | ✅                | ❌              | ❌    | ❌         |
+	 * | Default imports  | ❌  | ❌              | ❌                | ✅              | ✅    | ❌         |
+	 * | `src`            | ❌  | ❌              | ❌                | ✅              | ✅    | ❌         |
+	 * | DLL builds       | ❌  | ❌              | ❌                | ❌              | ❌    | ✅         |
```

Comment:
Do we have something like this in non-API docs? It looks useful.

Reply:
We don't, and I'm not sure if we should. Because we only promote using NIM (and ways to optimize it), the average developers should not be aware of other methods unless they do something wrong and get this error.

I don't have a strong opinion on this, so feel free to use it in order places if you want.

Reply:
> the average developers should not be aware of other methods unless they do something wrong and get this error.

That's a good point. I agree 👍